heron/tmaster/src/cpp/manager/tmetrics-collector.cpp

```diff
@@ -104,12 +106,34 @@ void TMetricsCollector::AddMetric(const PublishMetrics& _metrics) {
   }
 }
 
-MetricResponse* TMetricsCollector::GetMetrics(const MetricRequest& _request) {
+MetricResponse* TMetricsCollector::GetMetrics(const MetricRequest& _request,
+                                              const proto::api::Topology* _topology) {
   MetricResponse* response = new MetricResponse();
+
   if (metrics_.find(_request.component_name()) == metrics_.end()) {
-    LOG(ERROR) << "GetMetrics request received for unknown component " << _request.component_name();
-    response->mutable_status()->set_status(proto::system::NOTOK);
-    response->mutable_status()->set_message("Unknown component");
+    bool component_exists = false;
+    for (int i = 0; i < _topology->spouts_size(); i++) {
+      if ((_topology->spouts(i)).comp().name() == _request.component_name()) {
+        component_exists = true;
```

Comment:
update via https://github.com/twitter/heron/pull/1336/commits/8ddd534675296dfe2032edd6bbf7b0d2c5650bc8. Can you have a look?